executor/write.go

```diff
@@ -980,6 +1028,9 @@ func (e *InsertValues) initDefaultValues(row []types.Datum, hasValue []bool, ign
 			// Just leave generated column as null. It will be calculated later
 			// but before we check whether the column can be null or not.
 			needDefaultValue = false
+			if !hasValue[i] {
```

Comment:
any test case for this check?

Reply:
Test L1279-L1281:
```SQL
create table t(a int auto_increment key, b int);
set SQL_MODE=NO_AUTO_VALUE_ON_ZERO;
insert into t (b) value (a+1);
```
If we don't reset `a` to `NULL`, `a` will be `0`.